salt/modules/mount.py

```diff
@@ -1262,3 +1263,99 @@ def is_mounted(name):
         return True
     else:
         return False
+
+
+def read_mount_cache(name):
+    '''
+    .. versionadded:: Oxygen
+
+    Provide information if the path is mounted
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' mount.read_mount_cache /mnt/share
+    '''
+    cache = salt.utils.mount.read_cache(__opts__)
+    if cache:
+        if 'mounts' in cache and cache['mounts']:
+            if name in cache['mounts']:
+                return cache['mounts'][name]
+    return {}
+
+
+def write_mount_cache(real_name,
+                      device,
+                      mkmnt,
+                      fstype,
+                      mount_opts):
+    '''
+    .. versionadded:: Oxygen
+
+    Provide information if the path is mounted
+
+    :param real_name:     The real name of the mount point where the device is mounted.
+    :param device:        The device that is being mounted.
+    :param mkmnt:         Whether or not the mount point should be created.
+    :param fstype:        The file system that is used.
+    :param mount_opts:    Additional options used when mounting the device.
+    :return:              Boolean if message was sent successfully.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' mount.write_mount_cache /mnt/share /dev/sda1 False ext4 defaults,nosuid
+    '''
+    cache = salt.utils.mount.read_cache(__opts__)
+
+    if cache:
+        if 'mounts' in cache:
+            cache['mounts'][real_name] = {'device': device,
+                                          'fstype': fstype,
+                                          'mkmnt': mkmnt,
+                                          'opts': opts}
```

Comment:
`opts` should be `mount_opts` now 

There are a couple other references below that also need to be adjusted.